jetty-webapp/src/main/java/org/eclipse/jetty/webapp/ClassMatcher.java

```diff
@@ -496,10 +496,16 @@ public ClassMatcher()
     {
     }
 
-    public ClassMatcher(ClassMatcher patterns)
+    public ClassMatcher(ClassMatcher matcher)
     {
-        if (patterns != null)
-            setAll(patterns.getPatterns());
+        if (matcher != null)
+        {
+            _entries.putAll(matcher._entries);
+            _patterns.include(matcher._patterns.getIncluded().toArray(new Entry[0]));
+            _patterns.exclude(matcher._patterns.getExcluded().toArray(new Entry[0]));
+            _locations.include(matcher._locations.getIncluded().toArray(new Entry[0]));
+            _locations.exclude(matcher._locations.getExcluded().toArray(new Entry[0]));
+        }
```

Comment:
The `ClassMatcher.add()` method actually sets `_locations` and `_patterns` anyway.

But I think this is equivalent behaviour, unless someone has a subclass of `ClassMatcher` to override `add(Entry)`.

Reply:
yeah, it was more to keep the static analysis quiet so we can get a clean test.
Hmmm but your point about add is valid.   I'll see if I can suppress the analysis.